src/ui/components/post/post_listing.rs

```diff
@@ -1,151 +1,308 @@
-    error.set(None);
-    match block_user_action.value().get() {
-      None => {}
-      Some(Ok(_o)) => {}
-      Some(Err(e)) => {
-        error.set(Some(e.to_string()));
-      }
-    }
-  });
+  let on_block_submit = move |ev: SubmitEvent| {
+    ev.prevent_default();
+
+    create_local_resource(
+      move || (),
+      move |()| async move {
+        let form = BlockPerson {
+          person_id: post_view.get().creator.id,
+          block: true,
+        };
+
+        let result = LemmyClient.block_user(form).await;
+
+        match result {
+          Ok(_o) => {}
+          Err(e) => {
+            error.set(Some(e));
+          }
+        }
+      },
+    );
+  };
+
+  let report_post_action = create_server_action::<ReportPostFn>();
+  let report_validation = create_rw_signal::<String>("".into());
 
-  let report_post_action = create_server_action::<BlockUserFn>();
+  let query = use_query_map();
+  let ssr_error = move || query.with(|params| params.get("error").cloned());
 
-  create_effect(move |_| {
-    error.set(None);
-    match report_post_action.value().get() {
-      None => {}
-      Some(Ok(_o)) => {}
-      Some(Err(e)) => {
-        error.set(Some(e.to_string()));
+  if let Some(e) = ssr_error() {
+    let le = serde_json::from_str::<LemmyAppError>(&e[..]);
+
+    match le {
+      Ok(e) => match e {
+        LemmyAppError {
+          error_type: LemmyAppErrorType::MissingReason,
+          content: c,
+        } => {
+          let id = format!("{}", post_view.get().post.id);
+          if c.eq(&id) {
+            report_validation.set("input-error".to_string());
+          }
+        }
+        _ => {
+          report_validation.set("".to_string());
+        }
+      },
+      Err(_) => {
+        logging::error!("error decoding error - log and ignore in UI?");
       }
     }
-  });
+  }
+
+  let reason = RwSignal::new(String::new());
+
+  let on_report_submit = move |ev: SubmitEvent| {
+    ev.prevent_default();
+
+    create_local_resource(
+      move || (),
+      move |()| async move {
+        let form = CreatePostReport {
+          post_id: post_view.get().post.id,
+          reason: reason.get(),
+        };
+
+        let result = try_report(form).await;
+
+        match result {
+          Ok(_o) => {}
+          Err(e) => {
+            error.set(Some(e.clone()));
+
+            let _id = format!("{}", post_view.get().post.id);
+
+            match e {
+              LemmyAppError {
+                error_type: LemmyAppErrorType::MissingReason,
+                content: _id,
+              } => {
+                report_validation.set("input-error".to_string());
+              }
+              _ => {
+                report_validation.set("".to_string());
+              }
+            }
+          }
+        }
+      },
+    );
+  };
 
   view! {
-    <tr>
-      <td class="flex flex-col text-center">
-        <ActionForm action=vote_action>
+    <tr class="flex sm:table-row">
```

Comment:
While fine for the time being, I think we should avoid using tables for things that aren't tabular data.

Reply:
i should have marked the PR as draft again before pushing update - this code may have been removed or changed